src/System.ComponentModel.TypeConverter/tests/DescriptorTestAttribute.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.ComponentModel.Tests
+{
+    [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
+    internal class DescriptorTestAttribute : Attribute
+    {
+        public string TestString;
+
+        public DescriptorTestAttribute(string testString = nameof(TestString))
+        {
+            TestString = testString;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return (obj as DescriptorTestAttribute)?.TestString == TestString;
+        }
+
+        public override int GetHashCode() => base.GetHashCode();
```

Comment:
Equals and GetHashCode should both be overriden - you're only overriding Equals. For GetHashCode you can return TestString.GetHashCode()